### ISSUE TITLE
Bytes 0.5: Hello world fails with "impl has extra requirement `B: hello_world::HelloRequest_MESSAGE::_bytes::BufMut`"

### ISSUE THREAD

**Issue by Ciantic** [hyperium/tonic#215]
## Bug Report
I followed Hello World, but updated the bytes to 0.5, and got this:

```powershell
error[E0276]: impl has stricter requirements than trait
 --> C:\Source\Rust\TonicHello\helloworld-tonic\target\debug\build\helloworld-tonic-4114755bd1738601\out/helloworld.rs:1:28
  |
1 | #[derive(Clone, PartialEq, ::prost::Message)]
  |                            ^^^^^^^^^^^^^^^^ impl has extra requirement `B: hello_world::HelloRequest_MESSAGE::_bytes::BufMut`

error[E0276]: impl has stricter requirements than trait
 --> C:\Source\Rust\TonicHello\helloworld-tonic\target\debug\build\helloworld-tonic-4114755bd1738601\out/helloworld.rs:1:28
  |
1 | #[derive(Clone, PartialEq, ::prost::Message)]
  |                            ^^^^^^^^^^^^^^^^ impl has extra requirement `B: hello_world::HelloRequest_MESSAGE::_bytes::Buf`
```

### Version

```
[package]
name = "helloworld-tonic"
version = "0.1.0"
edition = "2018"

[[bin]] # Bin to run the HelloWorld gRPC server
name = "helloworld-server"
path = "src/server.rs"


[dependencies]
tonic = "0.1.0-beta.1"
bytes = "0.5"
prost = "0.5"
tokio = { version = "0.2", features = ["macros"] }

[build-dependencies]
tonic-build = "0.1.0-beta.1"
```

(It works just fine if I downgrade bytes to 0.4)

**Comment by LucioFranco** [comment]
@Ciantic Yeah, currently, waiting on `prost` to update to `bytes 0.5` which should come in the next release. So for now you'll need to include bytes 0.4. This is the reason we have a `beta.1` instead of a `0.1`

**Comment by LucioFranco** [comment]
This should be fixed on master and will be fixed in the upcomming 0.1 release.
